phd-tests/framework/src/guest_os/mod.rs

```diff
@@ -35,6 +46,16 @@ pub(super) trait GuestOs {
 
     /// Indicates whether the guest has a read-only filesystem.
     fn read_only_fs(&self) -> bool;
+
+    /// Some guests need to amend incoming shell commands from tests in order to
+    /// get output to display on the serial console in a way those guests can
+    /// accept (e.g. by clearing the screen immediately before running each
+    /// command). If a guest requires this kind of amendment for a given
+    /// command, this function returns `Some(amended_command)`. Otherwise it
+    /// returns `None`.
+    fn amend_shell_command(&self, _cmd: &str) -> Option<String> {
+        None
+    }
```

Comment:
nit, take it or leave it: why not just have this be
```suggestion
    fn amend_shell_command<'a>(&self, cmd: &'a str) -> Cow<'a, str> {
        Cow::Borrowed(cmd)
    }
```

rather than having to bother with an `Option`?